src/storage/src/hummock/sstable_store.rs

```diff
@@ -471,6 +438,18 @@ pub trait SstableStoreWrite: Send + Sync {
         data: Bytes,
         policy: CachePolicy,
     ) -> HummockResult<()>;
+
+    async fn create_put_sst_stream(
+        &self,
+        sst_id: HummockSstableId,
+        policy: CachePolicy,
+    ) -> HummockResult<SstableStreamingUploader>;
+
+    async fn finish_put_sst_stream(
```

Comment:
I think `meta` is not part of the sst data stream, and should not be included when we finish the streaming upload. 

Besides, instead of introducing a `finish_put_sst_stream` in the `SstableStoreWrite` trait, we can add a method `fn finish(self)` in the `SstableStreamingUploader` so that the uploader itself can finish the upload. 

To enable `SstableStreamingUploader` to `finish` by itself, we may store a reference to `SstableStore` in `SstableStreamingUploader`. In order to make the lifetime of `SstableStreamingUploader`, the reference to `SstableStore` will be `Arc<SstableStore>`, and then `create_put_sst_stream` will take `self: Arc<Self>` instead of `&self`.

Reply:
`self: Arc<Self>` will take ownership of `Arc<Self>`, so if I want to call other methods of `SstableStoreRef` after `create_put_sst_stream`, I will have write `sstable_store.clone().create_put_sst_stream(...);`